sources/web/datalab/polymer/test/file-browser-test.ts

```diff
@@ -34,43 +34,48 @@ window.addEventListener('WebComponentsReady', () => {
       return [new MockFile('', path)];
     }
   }
-  public pathToPathHistory(path: string): DatalabFile[] {
-    return [new MockFile('', path)];
-  }
-}
-
-describe('<file-browser>', () => {
-  let testFixture: FileBrowserElement;
-  const startuppath = new DatalabFileId('testpath', FileManagerType.MOCK);
-
-  const mockFiles = [
-    new MockFile('file1'),
-    new MockFile('file2'),
-    new MockFile('file3'),
-  ];
-
-  const fileAddButtonIds = [
-    'addNotebookButton',
-    'addFileButton',
-    'addFolderButton',
-  ];
-  const fileOpsButtonIds = [
-    'uploadButton',
-    'editAsTextButton',
-    'copyButton',
-    'moveButton',
-    'renameButton',
-    'deleteButton',
-  ];
-
-  before(() => {
-    SettingsManager.getUserSettingsAsync = (forceRefresh: boolean) => {
-      assert(forceRefresh === true, 'file-browser should refresh settings on load');
-      const mockSettings: common.UserSettings = {
-        idleTimeoutInterval: '',
-        idleTimeoutShutdownCommand: '',
-        startuppath: startuppath.path,
-        theme: 'light',
+
+  describe('<file-browser>', () => {
+    let testFixture: FileBrowserElement;
+    const startuppath = new DatalabFileId('testpath', FileManagerType.MOCK);
+
+    const mockFiles = [
+      new MockFile('file1'),
+      new MockFile('file2'),
+      new MockFile('file3'),
+    ];
+
+    const alwaysEnabledButtonIds = [
+      'newNotebookButton',
+      'newFileButton',
+      'newFolderButton',
+      'uploadButton',
+    ];
+    const singleSelectionEnabledButtonIds = [
+      'editAsTextButton',
+      'copyButton',
+      'moveButton',
+      'renameButton',
+    ];
+    const multiSelectionEnabledButtonIds = [
+      'deleteButton',
+    ];
+
+    function assertEnabledState(id: string, enabled: boolean) {
+      const button = testFixture.$[id] as HTMLElement;
+      assert(button.hasAttribute('disabled') === !enabled, id + ' is expected to be enabled');
```

Comment:
This message is not correct when enabled==false.

Reply:
Done.